### PR TITLE
Rating system mutations

### LINKED ISSUE #88: Rating system graphql mutations
Create and delete graphql mutations for creating or deleting rating system.
- [ ] CreateRatingSystem mutation
- [ ] GetRatingSystem mutation

### PR DESCRIPTION
**Pull request by christopherbarnett**
In this PR, 
- Created graphql schema for rating systems
- Created a get rating system query and create rating system mutation
- Created rating system controller 
- Updated dependencies 
- Made unit tests that are all passing

Query and mutation both seem to be functioning as expected on localhost:8080/graphiql

Closes #88

### CODE REVIEW

File: server/src/main/java/com/github/animelist/animelist/AnimeListRuntimeWiringConfigurer.java [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 import graphql.scalars.ExtendedScalars;
 import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.TypeRuntimeWiring;
 import org.springframework.graphql.execution.RuntimeWiringConfigurer;
 import org.springframework.stereotype.Component;
 
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring;
+
 @Component
 public class AnimeListRuntimeWiringConfigurer implements RuntimeWiringConfigurer {
```
Comment (jvoucsb) [inline server/src/main/java/com/github/animelist/animelist/AnimeListRuntimeWiringConfigurer.java:14]: We need type resolver for the interface "RatingSystem" that was defined in the schema

File: server/src/main/java/com/github/animelist/animelist/controller/RatingSystemController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return ratingSystemService.getRatingSystem(ratingSystemID).orElseThrow();
+    }
+
+    @MutationMapping
+    @PreAuthorize("isAuthenticated()")
+    public RatingSystem createRatingSystem(@Argument("input") final RatingSystemInput input){
+        JwtUserDetails userDetails = getUserDetails();
+
+        final RatingSystem ratingSystem;
+        if (input.ratingSystemType() == RatingSystemType.DISCRETE) {
```
Comment (jvoucsb) [inline server/src/main/java/com/github/animelist/animelist/controller/RatingSystemController.java:40]: style change can be implemented using the switch pattern matching in newer versions of java
```
return switch(input.ratingSystemType()) {
    case DISCRETE -> DiscreteRatingSystem.builder()...
    case CONTINUOUS -> ContinuousRatingSystem.builder()...
}
```

**Review by jvoucsb — APPROVED** [review]
good good looks good excellent

**Review by AHuangHe — APPROVED** [review]
LGTM!

**Pull request merged**